1.8/administration/access-node/tunnel.md

```diff
@@ -0,0 +1,118 @@
+A SRV DNS record is a mapping from a name to a IP/port pair. DC/OS creates SRV records in the form `_<port-name>._<service-name>._tcp.marathon.mesos`. The HTTP proxy exposes these as URLs. This feature can be useful for communicating with DC/OS services.
+
+# VPN
+DC/OS Tunnel provides you with full access to the DNS, masters, and agents from within the cluster. OpenVPN requires root privileges to configure these routes.
+
+# Using DC/OS Tunnel
+
+## Prerequisites
+* DC/OS 1.8+
+* Docker
+* Requires [SSH access](/1.8/administration/access-node/sshcluster/) (key authentication only).
+* For VPN functionality, you must have [the OpenVPN client](https://openvpn.net/index.php/open-source/downloads.html) installed.
+* The DC/OS Tunnel package must be installed: run `dcos package install tunnel-cli --cli` from the DC/OS CLI.
+
+**Note:** Only Linux and OS X are supported currently.
+
+##  SOCKS
+Run the following command from the DC/OS CLI:
+
+```
+$ dcos tunnel socks
+```
+
+Configure your application to use the proxy on port 1080: `127.0.0.1:1080`
+
+### Port Forwarding
+The SOCKS proxy works by port forwarding. Append `.mydcos.directory` to the end of your domain when you enter commands. For instance, `http://example.com:1080/?query=hello` becomes `http://example.com.mydcos.directory:1080/?query=hello`.
+
+##  HTTP
+Run the following command from the DC/OS CLI:
+
+```
+$ sudo dcos tunnel http
+```
+
+To run the HTTP proxy without root privileges, use the `--port` flag to configure it to use another port:
+
+```
+$ dcos tunnel http --port 8000
+```
+
+Then, configure your application to run HTTP on the port you specified above.
+
+### Port Forwarding
+The HTTP proxy works by port forwarding. Append `.mydcos.directory` to the end of your domain when you enter commands. For instance, `http://example.com:8080/?query=hello` becomes `http://example.com.mydcos.directory:8080/?query=hello`.
+
+### SRV Records
+The HTTP proxy exposes DC/OS SRV records as URLs in the form `_<port-name>._<service-name>._tcp.marathon.mesos.mydcos.directory`.
+
+To name a port, add `name` to the `portMappings` or `portDefinitions` field of a Marathon application definition. Whether you use `portMappings` or `portDefinitions` depends on whether you are using `BRIDGE` or `HOST` networking. [Learn more about networking and ports in Marathon](https://mesosphere.github.io/marathon/docs/ports.html).
+
+```json
+"portMappings": [
+    {
+        "name": "<my-port-name>",
+        "containerPort": 3000,
+        "hostPort": 0,
+        "servicePort": 10000,
+        "labels": {
+             "VIP_0": "1.1.1.1:30000"
+        }
+    }
+]
+```
+
+```json
+"portDefinitions": [
+    {
+      "name": "<my-port-name>",
+      "protocol": "tcp",
+      "port": 0,    
+    }
+  ]
+```
+
+Alternatively, you can add a named port from the DC/OS web interface. Go to the **Services** tab, click the name of your service, and then click **Edit**. Enter a name for your port on the **Network** tab.
```

Comment:
Maybe we should make the UI way the primary way and the json way the alternative way?